File: app/src/main/java/com/codepath/debuggingchallenges/activities/MoviesActivity.java
package com.codepath.debuggingchallenges.activities;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.RecyclerView;

import com.codepath.debuggingchallenges.R;
import com.codepath.debuggingchallenges.adapters.MoviesAdapter;
import com.codepath.debuggingchallenges.models.Movie;
import com.loopj.android.http.AsyncHttpClient;
import com.loopj.android.http.JsonHttpResponseHandler;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

import cz.msebera.android.httpclient.Header;

public class MoviesActivity extends AppCompatActivity {

    private static final String API_KEY = "a07e22bc18f5cb106bfe4cc1f83ad8ed";

    RecyclerView rvMovies;
    MoviesAdapter adapter;
    ArrayList<Movie> movies;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_movies);
        rvMovies = findViewById(R.id.rvMovies);

        // Create the adapter to convert the array to views
        MoviesAdapter adapter = new MoviesAdapter(movies);

        // Attach the adapter to a ListView
        rvMovies.setAdapter(adapter);

        fetchMovies();
    }


    private void fetchMovies() {
        String url = " https://api.themoviedb.org/3/movie/now_playing?api_key=";
        AsyncHttpClient client = new AsyncHttpClient();
        client.get(url, null, new JsonHttpResponseHandler() {
            @Override
            public void onSuccess(int statusCode, Header[] headers, JSONObject response) {
                try {
                    JSONArray moviesJson = response.getJSONArray("results");
                    movies = Movie.fromJSONArray(moviesJson);
                } catch (JSONException e) {
                    e.printStackTrace();
                }
            }
        });
    }
}

File: app/src/main/java/com/codepath/debuggingchallenges/activities/MoviesActivity.kt
package com.codepath.debuggingchallenges.activities

import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.RecyclerView
import com.codepath.debuggingchallenges.adapters.MoviesAdapter
import android.os.Bundle
import android.util.Log
import androidx.recyclerview.widget.LinearLayoutManager
import com.codepath.debuggingchallenges.R
import com.codepath.asynchttpclient.AsyncHttpClient
import com.codepath.asynchttpclient.callback.JsonHttpResponseHandler
import org.json.JSONException
import com.codepath.debuggingchallenges.models.Movie
import okhttp3.Headers
import java.util.ArrayList

private const val TAG = "MoviesActivity"
class MoviesActivity : AppCompatActivity() {
    private var rvMovies: RecyclerView? = null
    var adapter: MoviesAdapter? = null
    //var movies: List<Movie>? = null
    private val movies = mutableListOf<Movie>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_movies)
        rvMovies = findViewById(R.id.rvMovies)

        // Create the adapter to convert the array to views
        adapter = MoviesAdapter(movies)

        rvMovies?.layoutManager = LinearLayoutManager(this)

        // Attach the adapter to a ListView
        rvMovies?.adapter = adapter
        fetchMovies()
    }

    private fun fetchMovies() {
        val url = " https://api.themoviedb.org/3/movie/now_playing?api_key=$API_KEY"
        val client = AsyncHttpClient()
        client[url, null, object : JsonHttpResponseHandler() {
            override fun onSuccess(statusCode: Int, headers: Headers, response: JSON) {
                try {
                    val moviesJson = response.jsonObject.getJSONArray("results")
                    //movies = Movie.fromJSONArray(moviesJson)
                    movies.addAll(Movie.fromJSONArray(moviesJson))
                    adapter?.notifyDataSetChanged()
                } catch (e: JSONException) {
                    e.printStackTrace()
                }
            }

            override fun onFailure(statusCode: Int, headers: Headers, response: String, throwable: Throwable) {
                Log.e(MoviesActivity::class.java.simpleName, "Error retrieving movies: ", throwable)
            }
        }]
    }

    companion object {
        private const val API_KEY = "a07e22bc18f5cb106bfe4cc1f83ad8ed"
    }
}